web/src/features/charts/elements/AreaGraphLayers.tsx

```diff
@@ -53,7 +53,7 @@ function AreaGraphLayers({
     .x((d: any) => timeScale(d.data.datetime))
     .y0((d) => valueScale(d[0]))
     .y1((d) => valueScale(d[1]))
-    .defined((d) => (shouldHideEmptyData ? d[1] > 0 : Number.isFinite(d[1])));
```

Comment:
Mind deleting this part a little further up as it's no longer used?
```ts
  const shouldHideEmptyData = isDataInteractive && layers.length > 1;
```

Reply:
Oh and the changes here causes these empty lines to re-appear:
<img width="437" alt="image" src="https://github.com/user-attachments/assets/656098c7-614b-424f-a0d2-d7d3c532a04d" />

I think we should just check that it's not 0 instead of removing this. If that makes sense.

The production app looks like this:
<img width="437" alt="image" src="https://github.com/user-attachments/assets/ba9212d5-319d-4423-b066-1a220a680d3d" />

Reply:
Excellent catch @VIKTORVAV99 !! Let me change this now

Reply:
Actually, when I change it to `!= 0`, then the negative storage graph is gone:

<img width="415" alt="image" src="https://github.com/user-attachments/assets/2cbc9007-d639-4ea5-9ddf-a628c5721f10" />


If the condition is removed (i.e. only non-finite points are removed), then the (negative) storage graph is back:

<img width="424" alt="image" src="https://github.com/user-attachments/assets/65b99b34-88ff-41dc-a053-b8a2bb75066b" />

Do you see this as well?

Reply:
Yeah I noticed the same. I didn't expect it to remove the whole thing.

I can test a few things locally as well.

Reply:
Does this work for you? It seems to work for the zones I have tested. 
```ts
    .defined((d) =>
      shouldHideEmptyData ? (d[1] > 0 ? d[1] != 0 : d[0] != 0) : Number.isFinite(d[1])
    );
```
Basically is just checks which end value it should use.

We can probably clean it up a bit though.

Reply:
yes that works. I rewrote it like this to:
1. check NaNs
2. remove points that start and end at the same place (y-axis)

```js
    .defined((d) => {
      if (!Number.isFinite(d[0]) || !Number.isFinite(d[1])) {
        return false;
      }
      if (shouldHideEmptyData && d[0] == d[1]) {
        // zero area height
        return false;
      }
      return true;
    });
```

My tests show that the above both hides solar values that equal zero, and shows (negative) storage values. What do you think?

Reply:
It looks like it works just as well and is a bit easier to understand, especially the equal part!

If we want to optimize it though we can do this though:
```ts
.defined((d) => 
  Number.isFinite(d[0]) && 
  Number.isFinite(d[1]) && 
  (!shouldHideEmptyData || d[0] !== d[1])
);
```

Reply:
I'll let you decide which one you prefer!

Reply:
Hehe maybe don't as I'll always opt for the most optimized version, it's a curse...

So maybe @tonypls can decide by the time we wake up tomorrow or we'll just merge one of them 😅

Reply:
As the charts are already quite complex, I'd lean towards the less optimized and easy to ready if statement implementation that mere mortals can read :)